frida-gum/src/range_details.rs

```diff
@@ -40,32 +40,43 @@ pub enum PageProtection {
 }
 
 /// The file association to a page.
+#[derive(Clone)]
 pub struct FileMapping<'a> {
-    file_mapping: *const gum_sys::GumFileMapping,
+    path: &'a str,
+    size: usize,
+    offset: u64,
     phantom: PhantomData<&'a gum_sys::GumFileMapping>,
 }
 
 impl<'a> FileMapping<'a> {
-    pub(crate) fn from_raw(file: *const gum_sys::GumFileMapping) -> Self {
-        Self {
-            file_mapping: file,
-            phantom: PhantomData,
+    pub(crate) fn from_raw(file: *const gum_sys::GumFileMapping) -> Option<Self> {
+        if file.is_null() {
+            None
+        } else {
+            Some(unsafe {
+                Self {
+                    path: CStr::from_ptr((*file).path).to_str().unwrap(),
```

Comment:
Will the path variable live long enough/not be freed at some point as well? Maybe a `clone()` here or even [to_string_lossy](https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.to_string_lossy) would be safer?